pkg/core/core.go

```diff
@@ -167,3 +189,105 @@ func GetBroadcastAddress(network *net.IPNet) (net.IP, error) {
 
 	return ip, nil
 }
+
+
+// Returns the net.IPMask necessary for the provided divisor.
+// Errors if address space if insufficient or division is not possible.
+func GetMaskWithDivisor(divisor int64, addressCount *big.Int, IPv4 bool) (net.IPMask, error) {
+	div := big.NewInt(divisor)
+	if addressCount.Cmp(div) == -1 || div.Cmp(big.NewInt(0)) == 0 {
+		return nil, fmt.Errorf("Cannot divide %d Addresses into %d divisions\n", addressCount, div)
+	}
+
+	addressPartition := new(big.Int).Div(addressCount, div)
+	two := big.NewInt(2)
+	exponent := big.NewInt(0)
+	for two.Cmp(addressPartition) <= 0 {
+		two.Lsh(two, 1)
+		exponent.Add(exponent, big.NewInt(1))
+	}
+	subnetPrefix := int(exponent.Int64())
+	bits := net.IPv6len * 8
+	if IPv4 {
+		bits = net.IPv4len * 8
+		if subnetPrefix > 30 {
+			return nil, fmt.Errorf("Address Space is insufficient for %d subnets\n", div)
+		}
+	}
+	if subnetPrefix > 126 {
+		return nil, fmt.Errorf("Address Space is insufficient for %d subnets\n", div)
+	}
+	return net.CIDRMask(bits-subnetPrefix, bits), nil
+
+}
+
+
+// Divides the given network into N smaller networks.
+// Errors if division is not possible.
+func DivideCidr(network *net.IPNet, divisor int64) ([]net.IPNet, error) {
+	isIPv4 := helper.IsIPv4Network(network)
+
+	addressCount := GetAddressCount(network)
+	cidrWack, err := GetMaskWithDivisor(divisor, addressCount, isIPv4)
```

Comment:
Is `Wack` an acronym of some sorts? Wouldn't a name like `cidrMask` be more clear?

Reply:
The "wack" is technically a "/" when talking about ip's and cidr masks verbally.
10.0.0.0 "wack" 24.  Yes, it's a bit a silly to use it as a variable name, but it's where my mind went. But yeah we can fix it, it's an easy change.